### PR TITLE
[FLINK-19098][format] Make row data converters public

### PR DESCRIPTION
**Pull request by crazyzhou**
## What is the purpose of the change

This pull request makes the rowdata converters public. That way third party projects can utilize the converter to serialize and deserialize between csv/json `JsonNode` and `Rowdata`.

## Brief change log

  - Refactor json/csv RowData runtime converters out to a class
  - Make the rowdata converter public
  - Add `static` to `RowDataToCsvConverter` 

## Verifying this change

This change is already covered by existing tests, such as `JsonRowDeserializationSchemaTest`, etc.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): **no**
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: **no**
  - The serializers: **no**
  - The runtime per-record code paths (performance sensitive): **no**
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn/Mesos, ZooKeeper: **no**
  - The S3 file system connector: **no**

## Documentation

  - Does this pull request introduce a new feature? **no**

### CODE REVIEW

**Comment by crazyzhou** [comment]
@wuchong @danny0405 Can you help review this PR?

**Review by wuchong — CHANGES_REQUESTED** [review]
Thanks for the contribution @crazyzhou . I left some comments.

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonToRowDataConverters.java [hunk 1]
```diff
@@ -0,0 +1,391 @@
...
+
+	/**
+	 * Timestamp format specification which is used to parse timestamp.
+	 */
+	private final TimestampFormat timestampFormat;
+
+	public JsonToRowDataConverters(
+		boolean failOnMissingField,
+		boolean ignoreParseErrors,
+		TimestampFormat timestampFormat) {
```
[the remark this answers is not in the document]
Comment (crazyzhou) [reply flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonToRowDataConverters.java]: Fixed this and also reverted the accidental change with the same indent issue in the constructor of `JsonRowDataDeserializationSchema`

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonToRowDataConverters.java [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+import static java.time.format.DateTimeFormatter.ISO_LOCAL_DATE;
+import static org.apache.flink.formats.json.TimeFormats.ISO8601_TIMESTAMP_FORMAT;
+import static org.apache.flink.formats.json.TimeFormats.ISO8601_TIMESTAMP_WITH_LOCAL_TIMEZONE_FORMAT;
+import static org.apache.flink.formats.json.TimeFormats.SQL_TIMESTAMP_FORMAT;
+import static org.apache.flink.formats.json.TimeFormats.SQL_TIMESTAMP_WITH_LOCAL_TIMEZONE_FORMAT;
+import static org.apache.flink.formats.json.TimeFormats.SQL_TIME_FORMAT;
+
+/** Tool class used to convert from {@link JsonNode} to {@link RowData}. **/
+@Internal
+public class JsonToRowDataConverters implements Serializable {
```
[the remark this answers is not in the document]
Comment (crazyzhou) [reply flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonToRowDataConverters.java:67]: Added `private static final long serialVersionUID = 1L;` in all the four `Serializable` converters.

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java [hunk 3]
```diff
@@ -149,316 +114,12 @@ public boolean equals(Object o) {
 			return false;
 		}
 		JsonRowDataDeserializationSchema that = (JsonRowDataDeserializationSchema) o;
-		return failOnMissingField == that.failOnMissingField &&
-				ignoreParseErrors == that.ignoreParseErrors &&
-				resultTypeInfo.equals(that.resultTypeInfo) &&
-				timestampFormat.equals(that.timestampFormat);
```
[the remark this answers is not in the document]
Comment (crazyzhou) [reply flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataDeserializationSchema.java]: Keep all the parameters as we had in these (De)SerializationSchema.

File: flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataSerializationSchema.java [hunk 4]
```diff
@@ -110,240 +85,11 @@ public boolean equals(Object o) {
 			return false;
 		}
 		JsonRowDataSerializationSchema that = (JsonRowDataSerializationSchema) o;
-		return rowType.equals(that.rowType) && timestampFormat.equals(timestampFormat);
+		return rowType.equals(that.rowType) && timestampFormat.equals(that.timestampFormat);
```
Comment (crazyzhou) [inline flink-formats/flink-json/src/main/java/org/apache/flink/formats/json/JsonRowDataSerializationSchema.java:88]: Fixed a small issue in this `equals` method.

**Review by wuchong — APPROVED** [review]
Thanks for the contribution @crazyzhou . LGTM. 

Please use "rebase" instead of "merge" next time, otherwise, it's hard to find what changes.

**Comment by crazyzhou** [comment]
> Thanks for the contribution @crazyzhou . LGTM.
> 
> Please use "rebase" instead of "merge" next time, otherwise, it's hard to find what changes.

Thanks for the review. I will take care next time

**Comment by wuchong** [comment]
Rebased.

**Comment by wuchong** [comment]
Merging...

**Pull request merged**